test/js/ui/marker.test.js

```diff
@@ -0,0 +1,105 @@
+'use strict';
+
+var test = require('tap').test;
+var jsdom = require('jsdom');
+
+var extend = require('../../../js/util/util').extend;
+var Map = require('../../../js/ui/map');
+var Marker = require('../../../js/ui/marker');
+var Popup = require('../../../js/ui/popup');
+
+function prepareDOM(html) {
+    html = html || '<!doctype html><html><body><div id="map"></div></body></html>';
+    if (typeof document !== 'undefined') {
+        return;
+    }
+    global.document = jsdom.jsdom(html);
+    global.window = global.document.defaultView;
+    global.navigator = {
+        userAgent: 'JSDOM'
+    };
+    global.HTMLElement = global.window.HTMLElement;
+
+}
```

Comment:
I was having issues with the functions run by the tests failing because document + window weren't in scope. I refactored to use the [`jsdom-global`](https://github.com/rstacruz/jsdom-global) module which provides a `cleanup` function that removes the created variables from the global scope and can be called when the tests are complete. Is this acceptable? 

Otherwise we can take the approach that [`mapbox-gl-directions`](https://github.com/mapbox/mapbox-gl-directions/blob/master/package.json#L16) uses (browserify + smokestack + tap) but that would require some refactoring of the test suite.